examples/learning/learn_associations.ipynb

```diff
@@ -0,0 +1,330 @@
+      "\n",
+      "In a perfect world, the PES rule could be applied to learn this mapping from examples. However, when two distinct inputs cause the same neurons to fire, their decoded values will depend on one another. This leads to difficulty when trying to store multiple indpendent associations in the same memory.\n",
+      "\n",
+      "To solve this problem, a vector-space analog of Oja's rule, dubbed Vector-Oja's rule (or simply _Voja's rule_) was proposed. In essence, this unsupervised learning rule makes neurons fire selectively in response to their input. When used in conjunction with properly-chosen intercepts (corresponding to the largest dot-product between pairs of inputs), this approach makes it possible to scalably learn new associations in a spiking network.\n",
+      "\n",
+      "Voja's rule works by moving the encoders of the active neurons toward the current input. This can be stated succinctly as,\n",
+      "\n",
+      "$$\n",
+      "\\Delta e_i = \\kappa a_i (x - e_i)\n",
+      "$$\n",
+      "\n",
+      "where $e_i$ is the encoder of the $i^{th}$ neuron, $\\kappa$ is a modulatory learning rate (positive to move towards, and negative to move away), $a_i$ is the filtered activity of the $i^{th}$ neuron, and $x$ is the input vector encoded by each neuron. To see how this is related to Oja's rule, substituting $e_i$ with the row of weights $W_i$, $x$ for the pre-synaptic activity vector $b$, and letting $s = 1 / a_i$ be a dynamic normalizing factor, gives\n",
+      "\n",
+      "$$\n",
+      "\\Delta W_i = \\kappa a_i (b - s a_i W_i)\n",
+      "$$\n",
+      "\n",
+      "which is the update rule for a single row using Oja. For more details, see [Learning large-scale heteroassociative memories in spiking neurons](http://compneuro.uwaterloo.ca/publications/voelker2014a.html).\n",
+      "\n",
+      "This notebook will lead the reader through a basic example of building a network that can store and recall new associations."
+     ]
+    },
+    {
+     "cell_type": "markdown",
+     "metadata": {},
+     "source": [
+      "## Step 1: Configure some example data\n",
+      "\n",
+      "First, we will setup some keys (inputs) and values (outputs) for our network to store and recall."
+     ]
+    },
+    {
+     "cell_type": "code",
+     "collapsed": false,
+     "input": [
+      "import numpy as np\n",
+      "import matplotlib.pyplot as plt\n",
+      "%matplotlib inline\n",
+      "\n",
+      "import nengo"
+     ],
+     "language": "python",
+     "metadata": {},
+     "outputs": [],
+     "prompt_number": null
+    },
+    {
+     "cell_type": "code",
+     "collapsed": false,
+     "input": [
+      "num_items = 5\n",
+      "\n",
+      "d_key = 2\n",
+      "d_value = 4\n",
+      "\n",
+      "np.random.seed(7)\n",
```

Comment:
Making a [`RandomState`](http://docs.scipy.org/doc/numpy/reference/generated/numpy.random.RandomState.html) instance is a bit safer than mucking with numpy's global random state, I think... at least, that's what we do to generate `rng` in the test suite.